### PR TITLE
Bump Node.js to 18.14.2

### LINKED ISSUE #1115: Upgrade Node.js to v18 or higher
### Description

- Node.js 버전을 18.x 혹은 그 이상으로 업그레이드합니다.

### Reasons for suggestion

- 현재 사용 중인 16.x 버전은 2023-09-11에 LTS 지원이 종료(end-of-life)됩니다.
  - https://github.com/nodejs/release#release-schedule
  - https://nodejs.org/en/blog/announcements/nodejs16-eol/

### Proposed solution

- 다음 LTS 버전인 18.x로 업그레이드합니다.

### References

_No response_

### PR DESCRIPTION
**Pull request by Dogdriip**
## Self Checklist

- [x] I wrote a PR title in **English**.
- [x] I added an appropriate **label** to the PR.
- [x] I wrote a commit message in **English**.
- [x] I wrote a commit message according to [**the Conventional Commits specification**](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] [I added the appropriate **changeset**](https://github.com/channel-io/bezier-react/blob/next-v1/CONTRIBUTING.md#add-a-changeset) for the changes.
- [ ] [Component] I wrote **a unit test** about the implementation.
- [ ] [Component] I wrote **a storybook document** about the implementation.
- [ ] [Component] I tested the implementation in **various browsers**.
  - Windows: Chrome, Edge, (Optional) Firefox
  - macOS: Chrome, Edge, Safari, (Optional) Firefox
- [ ] [*New* Component] I added my username to the correct directory in the `CODEOWNERS` file.

## Related Issue

- Closes #1115

## Summary

- Node.js 버전을 `16.2.0`에서 `18.14.2`로 올립니다.

## Details

- Node.js 버전 업그레이드
  - `.nvmrc`, `package.json:engines`의 Node.js 버전을 수정합니다.
  - CI 환경(`.circleci/config.yml`, `.github/workflows/*`)의 Node.js 버전을 수정합니다.
- `ts-node` 버전 업그레이드
  - `8.10.2`에서 `10.9.1`로 올립니다.
- 작동 확인
  - [`yarn cache clean --all`](https://yarnpkg.com/cli/cache/clean) 후 install, lint, typecheck, build 등 대부분의 동작이 이상 없음을 확인했습니다.

## Breaking change or not (Yes/No)

No.

## References

- #1115
- https://nodejs.org/en/blog/announcements/v18-release-announce/
- https://www.npmjs.com/package/ts-node

### CODE REVIEW

**Review by sungik-choi — APPROVED** [review]
💯

**Review by inhibitor1217 — APPROVED** [review]

**Review by carlos-channel-io — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ defaults: &defaults
   working_directory: *workspace_root
 
   docker:
-    - image: node:16.2.0
+    - image: node:18.14.2
```
Comment (Dogdriip) [inline .circleci/config.yml:19]: Ref: https://hub.docker.com/_/node

File: .github/workflows/prerelease.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ jobs:
       - name: Setup Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: 16.x
+          node-version: 18
```
Comment (Dogdriip) [inline .github/workflows/prerelease.yml:25]: Ref: https://github.com/actions/setup-node#supported-version-syntax

**Comment by Dogdriip** [comment]
이거 Codecov 왜 낮아졌는지 모르겠네요 😓 @sungik-choi 머지해주실 수 있으실까요?

**Pull request merged**